tasks/chain/economics.go

```diff
@@ -54,11 +54,19 @@ type ChainEconomics struct {
 // Run starts processing batches of tipsets until the context is done or
 // an error occurs.
 func (p *ChainEconomics) Run(ctx context.Context) error {
+	node, closer, err := p.opener.Open(ctx)
+	if err != nil {
+		return xerrors.Errorf("open lens: %w", err)
+	}
+	defer closer()
```

Comment:
how does this relate to the lifetime of the API?
if the api fails, this whole task will need to restart, right?

Reply:
It does. But tasks are designed to be restarted by the scheduler. When processBatch returns an error then `wait.RepeatUntil` will return and closer will be called. For the lotus api that closes the rpc client so next time the task starts (after a 1 minute delay by default) it will acquire a new connection to the lotus api. 

The repo and sql lenses manage a single reference to the repo and calling closer here is a noop.